MaterialLib/Fluid/FluidProperties/FluidProperties.h

```diff
@@ -0,0 +1,111 @@
+/**
+ *  \copyright
+ *   Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *              Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   FluidProperties.h
+ *
+ * Created on November 29, 2016, 2:23 PM
+ */
+
+#ifndef OGS_FLUID_PROPERTIES_H
+#define OGS_FLUID_PROPERTIES_H
+
+#include <array>
+#include <memory>
+
+#include "MaterialLib/Fluid/FluidProperty.h"
+#include "MaterialLib/Fluid/PropertyVariableType.h"
+
+namespace MaterialLib
+{
+namespace Fluid
+{
+/// Fluid property type.
+enum class FluidPropertyType
+{
+    Density = 0,
+    Vicosity = 1,
+    HeatCapacity = 2,
+    ThermalConductivity = 3,
+    number_of_property_types = 4  ///< Number of property types.
+};
+
+const unsigned FluidPropertyTypeNumber =
+    static_cast<unsigned>(FluidPropertyType::number_of_property_types);
+
+/// Base class of fluid properties.
+class FluidProperties
+{
+public:
+    typedef std::array<double, PropertyVariableNumber> ArrayType;
+
+    FluidProperties(
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>&& density,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>&& viscosity,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>&& heat_capacity,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>&&
+            thermal_conductivity)
+    /* Will use this if MS visual studio compiler supports
```

Comment:
The latest MSVC compiler still does not support list initialization inside member initializer list or non-static data member initializer. Therefore I added ifdef MSVC for that.